components/web_discovery/browser/rsa.cc

```diff
@@ -59,33 +50,23 @@ std::unique_ptr<ImportedRSAKey> ImportRSAKey(
     return nullptr;
   }
 
-  auto key_pair = crypto::RSAPrivateKey::CreateFromPrivateKeyInfo(*decoded_key);
+  auto key_pair = crypto::keypair::PrivateKey::FromPrivateKeyInfo(*decoded_key);
   if (!key_pair) {
     return nullptr;
   }
 
-  std::vector<uint8_t> encoded_public_key;
-  if (!key_pair->ExportPublicKey(&encoded_public_key)) {
-    return nullptr;
-  }
-
-  auto info = std::make_unique<ImportedRSAKey>();
-
-  info->private_key = std::move(key_pair);
-  info->public_key_b64 = base::Base64Encode(encoded_public_key);
+  std::vector<uint8_t> encoded_public_key = key_pair->ToSubjectPublicKeyInfo();
 
-  return info;
+  return std::make_unique<ImportedRSAKey>(
+      std::move(*key_pair), base::Base64Encode(encoded_public_key));
```

Comment:
`ImportedRSAKey` seems to be duplicating in memory the information that we already have in `crypto::keypair::PrivateKey`. Ideally it is better to avoid these unnecessary conversion to string, or having our own custom types, but rather to use the type provided upstream, because we are paying in memory, performance, and potentially making the code harder to understand too.

Reply:
Unless Im misunderstanding something, for all effects and purposes, it looks like `ImportedRSAKey` holds the very same information that `crypto::keypair::PrivateKey` is meant to have, but in a slightly different format, so ultimately they do seem the same, but I could be missing something... let's take this conversation to a follow up PR as you suggested.